File: Java/SquareSide/app/src/main/java/com/martiandeveloper/squareside/MainActivity.java
package com.martiandeveloper.squareside;

import android.os.Bundle;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private MyView myView;
    private MyViewDefault myViewDefault;

    private FrameLayout activityMainDefaultContainerFL, activityMainContainerFL;
    private EditText activityMainNumberET;

    private float defaultSide;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        activityMainDefaultContainerFL = findViewById(R.id.activity_main_defaultContainerFL);
        activityMainContainerFL = findViewById(R.id.activity_main_containerFL);
        activityMainNumberET = findViewById(R.id.activity_main_numberET);

        findViewById(R.id.activity_main_drawMBTN).setOnClickListener(v -> {

            if ((activityMainNumberET.getText() != null && !activityMainNumberET.getText().toString().equals(""))) {

                try {

                    if (Float.parseFloat(activityMainNumberET.getText().toString()) >= 0.5 && Float.parseFloat(activityMainNumberET.getText().toString()) <= 4) {

                        ((InputMethodManager) getSystemService(INPUT_METHOD_SERVICE)).hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);

                        if (myViewDefault == null) {
                            defaultSide = Float.parseFloat(activityMainNumberET.getText().toString()) * 40F;

                            myViewDefault = new MyViewDefault(MainActivity.this, defaultSide);
                            activityMainDefaultContainerFL.addView(myViewDefault);
                        } else {
                            if (myView != null) activityMainContainerFL.removeView(myView);

                            myView = new MyView(MainActivity.this, Float.parseFloat(activityMainNumberET.getText().toString()) * defaultSide);
                            activityMainContainerFL.addView(myView);
                        }

                    } else
                        Toast.makeText(MainActivity.this, "Please enter a number between 0.5 and 4", Toast.LENGTH_SHORT).show();

                } catch (Exception e) {
                    Toast.makeText(MainActivity.this, "Please enter a number between 0.5 and 4", Toast.LENGTH_SHORT).show();
                }


            }

        });

    }

}

File: Kotlin/SquareSide/app/src/main/java/com/martiandeveloper/squareside/MainActivity.kt
package com.martiandeveloper.squareside

import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProviders
import com.martiandeveloper.squareside.databinding.ActivityMainBinding
import com.martiandeveloper.squareside.tools.EventObserver

class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: MainViewModel

    private lateinit var binding: ActivityMainBinding

    private lateinit var myView: MyView

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.let {
            it.viewModel = viewModel
            it.lifecycleOwner = this
        }

        observe()

        myView = MyView(this,200F)

        binding.activityMainContainerFL.addView(myView)

    }

    private fun observe() {

        with(viewModel) {

            viewModel.drawMBTNClick.observe(this@MainActivity, EventObserver {

                if (!number.value.isNullOrEmpty())

                    if (number.value!!.toFloat() in 0.5..4.0) {
                        (getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager).hideSoftInputFromWindow(
                            (currentFocus ?: View(this@MainActivity)).windowToken,
                            0
                        )

                        binding.activityMainContainerFL.removeView(myView)
                        myView = MyView(this@MainActivity,(number.value!!.toFloat().times(200)))
                        binding.activityMainContainerFL.addView(myView)

                    } else Toast.makeText(
                        this@MainActivity,
                        "Please enter a number between 0.5 and 4",
                        Toast.LENGTH_SHORT
                    ).show()



            })

        }

    }

}
